internal/image.go

```diff
@@ -29,13 +30,19 @@ func NewImage(pdf gofpdf.Pdf, math Math) *image {
 }
 
 // AddFromFile open an image from disk and add to PDF
-func (s *image) AddFromFile(path string, marginTop float64, indexCol float64, qtdCols float64, colHeight float64, prop props.Rect) {
+func (s *image) AddFromFile(path string, marginTop float64, indexCol float64, qtdCols float64, colHeight float64, prop props.Rect) (err error) {
 	info := s.pdf.RegisterImageOptions(path, gofpdf.ImageOptions{
 		ReadDpi:   false,
 		ImageType: "",
 	})
 
+	if info == nil {
+		err = errors.New("Could not register image options, maybe path/name is wrong")
+		return
+	}
```

Comment:
In my opinion, this best-guess attempt at giving a meaningful error message to the user is a good example of why "delaying" errors is actually worse than following Go idiomatic error handling approach :-)

I think that at least in this case we can do better, we can use gofpdf error state, something like (untested):

```
if info == nil {
    return s.pdf.Error()
}
```

this is less verbose and gives the best possible error back to the user.

Reply:
Ok, but in this case the panic occurs because the info is nil and I access info.Width(), isn't and error from gofpdf.